openlattice.yaml

```diff
@@ -1442,6 +1442,95 @@ paths:
               schema:
                 type: integer
             
+  ###################
+  ## DATA SET APIs ##
+  ###################
+  
+  /datastore/organization-database/{organizationId}/external-database-table:
+    get:
+      summary: Gets all OrganizationExternalDatabaseTable objects for an organization
+      operationId: getExternalDatabaseTables
+      tags: [dataset]
+      security:
+        - openlattice_auth: []
+        - http_auth: []
+      parameters:
+        - name: organizationId
+          in: path
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/OrganizationExternalDatabaseTable"
+  /datastore/organization-database/{organizationId}/external-database-table/external-database-column:
+    get:
+      summary: Gets a map of all OrganizationExternalDatabaseTable objects to OrganizationExternalDatabase columns that are contained within each table.
+      operationId: getExternalDatabaseTablesWithColumns
+      tags: [dataset]
+      security:
+        - openlattice_auth: []
+        - http_auth: []
+      parameters:
+        - name: organizationId
+          in: path
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        "200":
+          description: Success
+          content:
+            application/json:
+              schema:
+                type: object
+                additionalProperties:
+                  type: array
```

Comment:
This returns a map, is that converted to an array somehow or am I missing something?

Reply:
Yeah, this is actually because this endpoint is not according to openapi-specs :-\ .

Whenever there's a dynamic key, those all go into `additionalProperties`, without specifying its type.  All keys are supposed to be strings.

So what this is saying is: it returns a map ('object' in openapi-lingo) where all values are an array of OrganizationExternalDatabaseColumn.  The keys probably won't be deserialized into a string in all api-clients. 

To fix this, the endpoint should change to return something like (pseudolanguage):
```
[
  {
    table: OrganizationExternalDatabaseTable,
    columns: Set(OrganizationExternalDatabaseColumn)
  },
  ...
]
```

But that would be an api-change, can't be fixed here...